shared/utility/actuation/InverseKinematics.hpp

```diff
@@ -86,46 +82,45 @@ namespace utility::actuation::kinematics {
         }
 
         // Apply offsets to Htf and clamp to limits
-        const Eigen::Matrix<Scalar, 3, 1> ankle_x   = Htf.matrix().template leftCols<1>().template head<3>();
-        const Eigen::Matrix<Scalar, 3, 1> ankle_y   = Htf.matrix().template middleCols<1>(1).template head<3>();
-        const Eigen::Matrix<Scalar, 3, 1> ankle_pos = Htf.translation();
+        const Eigen::Matrix<Scalar, 3, 1> ankle_x = Htf.matrix().template leftCols<1>().template head<3>();
+        const Eigen::Matrix<Scalar, 3, 1> ankle_y = Htf.matrix().template middleCols<1>(1).template head<3>();
+        Eigen::Matrix<Scalar, 3, 1> rFTt          = Htf.translation();
 
-        const Eigen::Matrix<Scalar, 3, 1> hip_offset(LENGTH_BETWEEN_LEGS * 0.5,
-                                                     HIP_OFFSET_X,
-                                                     DISTANCE_FROM_BODY_TO_HIP_JOINT);
-        Eigen::Matrix<Scalar, 3, 1> target_leg = ankle_pos - hip_offset;
+        const Eigen::Matrix<Scalar, 3, 1> hip_offset(model.leg.LENGTH_BETWEEN_LEGS * 0.5,
+                                                     model.leg.HIP_OFFSET_X,
+                                                     model.leg.HIP_OFFSET_Z);
+        rFTt = rFTt - hip_offset;
 
-        const Scalar max_leg_length = UPPER_LEG_LENGTH + LOWER_LEG_LENGTH;
-        Scalar length               = target_leg.norm();
+        const Scalar max_leg_length = model.leg.UPPER_LEG_LENGTH + model.leg.LOWER_LEG_LENGTH;
+        Scalar length               = rFTt.norm();
         if (length > max_leg_length) {
-            target_leg = target_leg * max_leg_length / length;
-            length     = target_leg.norm();
+            rFTt = rFTt * max_leg_length / length;
         }
 
         // Knee pitch
-        const Scalar sqr_length    = length * length;
-        const Scalar sqr_upper_leg = UPPER_LEG_LENGTH * UPPER_LEG_LENGTH;
-        const Scalar sqr_lower_leg = LOWER_LEG_LENGTH * LOWER_LEG_LENGTH;
-        const Scalar cos_knee =
-            (sqr_upper_leg + sqr_lower_leg - sqr_length) / (2.0 * UPPER_LEG_LENGTH * LOWER_LEG_LENGTH);
-        const Scalar knee_pitch = M_PI - std::acos(std::max(std::min(cos_knee, Scalar(1.0)), Scalar(-1.0)));
+        const Scalar sqr_length    = std::pow(length, 2);
+        const Scalar sqr_upper_leg = std::pow(model.leg.UPPER_LEG_LENGTH, 2);
+        const Scalar sqr_lower_leg = std::pow(model.leg.LOWER_LEG_LENGTH, 2);
```

Comment:
`std::pow(x, 2)` is less performant than x*x since one uses a power function while the other is simply multiplication. Sometimes it doesn't matter but generally I'd use multiplication to square over powers